### ISSUE TITLE
How to train batch_normalization layers ? 

### ISSUE THREAD

**Issue by JackHenry1992** [DrSleep/tensorflow-deeplab-resnet#132]
I want to know how to train batch_normalization layers? 
Does it work if I just set `is_training=True` ? 
In your code ,  **_all_trainable=[v for v in tf.trainable_variables() if 'beta' not in v.name and 'gamma' not in v.name]_**.  Does it needed to contain 'beta & gamma' in all_trainable? 
Does it needed to apply_gradients to this `beta & gamma` value?

Can you provide some code for training BN layers?
Thanks for sharing your perfect work.

**Comment by Ariel-JUAN** [comment]
@JackHenry1992  Now I am working with finetuning the FCRN work, but the result is not good. It can not converge. Can you share the finetune code? My email is muamuawei@gmail.com  Please help me, thank you! please~  thankyou

**Comment by Ariel-JUAN** [comment]
Please, I really need your help. I know it is kind of reckless, but I stuck with finetuing the code. Please help me, thank you.

**Comment by JackHenry1992** [comment]
Hi @Ariel-JUAN , I just use the samilar code with this project, can you described your detail-problems?

**Comment by Ariel-JUAN** [comment]
What code are you using?
The loss curve is just like this:
![1](https://user-images.githubusercontent.com/30546375/32585556-f99629d2-c538-11e7-84b8-4309ebc9126e.png)

it can't converge. I am kind of misery==

**Comment by Ariel-JUAN** [comment]
If you can share your code, I will be really thank you.

**Comment by John1231983** [comment]
@JackHenry1992: Did you find the solution? Could you share with us?

**Comment by mokii** [comment]
Same Question! Does it needed to contain 'beta & gamma' in all_trainable? It must be Yes.
But is it OK to just apply_gradients to this beta & gamma value?
